### ISSUE TITLE
dbt clean regression

### ISSUE THREAD

**Issue by tmastny** [fishtown-analytics/dbt#2620]
### Describe the bug
In dbt 0.16.1 `dbt clean` fails without a profile: 

```bash
(dbt) dbt$ dbt --version
installed version: 0.16.1
   latest version: 0.17.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation
(dbt) dbt$ dbt clean
Running with dbt=0.16.1
Encountered an error while reading the project:
  ERROR: Runtime Error
  Could not find profile named 'profile'
Encountered an error:
Runtime Error
  Could not run dbt
```

In dbt 0.15.1, `dbt clean` works.

```bash
(dbt) dbt$ dbt --version
installed version: 0.15.1
   latest version: 0.17.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation
(dbt) dbt$ dbt clean
Running with dbt=0.15.1
Checking target/*
 Cleaned target/*
Finished cleaning all paths.
```

### Steps To Reproduce
Delete any profile found in `~/.dbt/profile.yml`. 

Install 0.16.1:
```bash
pip install dbt==0.16.1
```
Navigate to dbt project:
```
dbt clean
```

Repeat for 0.15.1 to confirm regression.

### Expected behavior
I expected `dbt clean` to work without a profile. This broke some of our automated jobs when we tried to upgrade.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [X] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
Multiple versions. See above.

**The operating system you're using:**
macOS 10.14.6

**The output of `python --version`:**
```
(dbt) dbt$ python --version
Python 3.7.3
```

### Additional context
Most people probably don't run `dbt clean` without a profile, but it was causing us confusion, so wanted to document it as a breaking change at least.

I also tested this with 0.17.0: same error as 0.16.1.

```
(dbt) dbt$ dbt --version
installed version: 0.17.0
   latest version: 0.17.0

Up to date!

Plugins:
  - bigquery: 0.17.0
  - snowflake: 0.17.0
  - redshift: 0.17.0
  - postgres: 0.17.0
(dbt) dbt$ dbt clean
Running with dbt=0.17.0
Encountered an error while reading the project:
  ERROR: Runtime Error
  Could not find profile named 'profile'
Encountered an error:
Runtime Error
  Could not run dbt
```

**Comment by jtcohen6** [comment]
Thanks for pointing this out, @tmastny. You're right that it's a regression.

We removed the profile requirement for `dbt deps` in 0.16.1 (https://github.com/fishtown-analytics/dbt/issues/2231), but it seems like we didn't fix it for `dbt clean`.

**Comment by jtcohen6** [comment]
Resolved by #2649
